feat: add v0 provider support

colesmcintosh opened the pull request:
## Title

Add v0 provider support to LiteLLM

## Relevant issues

N/A - New feature

## Pre-Submission checklist

**Please complete all items before asking a LiteLLM maintainer to review your PR**

- [x] I have Added testing in the [`tests/litellm/`](https://github.com/BerriAI/litellm/tree/main/tests/litellm) directory, **Adding at least 1 test is a hard requirement** - [see details](https://docs.litellm.ai/docs/extras/contributing_code)
- [x] I have added a screenshot of my new test passing locally 
- [x] My PR passes all unit tests on [`make test-unit`](https://docs.litellm.ai/docs/extras/contributing_code)
- [x] My PR's scope is as isolated as possible, it only solves 1 specific problem


## Type

🆕 New Feature

## Changes

### Summary
This PR adds support for the v0 provider (https://v0.dev) to LiteLLM. v0 is an AI-powered UI generation tool that provides OpenAI-compatible chat completions API.

### Implementation
- Added v0 as a new OpenAI-compatible provider in `litellm/llms/v0/`
- Registered v0 in provider lists and constants
- Added model configurations for all three v0 models:
  - `v0/v0-1.0-md` - Legacy model with 128k context
  - `v0/v0-1.5-md` - Medium model with 128k context  
  - `v0/v0-1.5-lg` - Large model with 512k context
- Configured token limits, pricing, and capabilities (including vision support)
- Added provider detection for `v0/` prefix and `https://api.v0.dev/v1` endpoint
- Created comprehensive unit tests in `tests/llm_translation/test_v0.py`

### Features Supported
The v0 provider inherits from `OpenAILikeChatConfig` and supports:
- ✅ Streaming responses
- ✅ Function calling and tools
- ✅ System messages
- ✅ Vision/multimodal inputs
- ✅ All standard OpenAI parameters

### Usage Example
```python
import litellm
import os

# Set your v0 API key
os.environ["V0_API_KEY"] = "your-v0-api-key"

# Use v0 models
response = litellm.completion(
    model="v0/v0-1.5-md",
    messages=[{"role": "user", "content": "Hello\!"}]
)

# With vision
response = litellm.completion(
    model="v0/v0-1.5-lg",
    messages=[{
        "role": "user",
        "content": [
            {"type": "text", "text": "What's in this image?"},
            {"type": "image_url", "image_url": {"url": "https://example.com/image.jpg"}}
        ]
    }]
)
```

### Testing
- Created 7 unit tests covering:
  - Provider initialization
  - API configuration
  - Provider detection logic
  - Model configuration validation
  - OpenAI parameter support
- All tests pass locally (see screenshot below)
- Note: v0 API requires a Premium or Team plan for actual API calls

### Test Results Screenshot
```
============================= test session starts ==============================
platform darwin -- Python 3.11.13, pytest-7.4.4, pluggy-1.5.0
rootdir: /Users/colesmcintosh/Projects/litellm-worktrees/feature-v0-models
collected 7 items

tests/llm_translation/test_v0.py::test_get_llm_provider_v0 PASSED        [ 14%]
tests/llm_translation/test_v0.py::test_v0_completion_call PASSED         [ 28%]
tests/llm_translation/test_v0.py::test_v0_config_initialization PASSED   [ 42%]
tests/llm_translation/test_v0.py::test_v0_get_openai_compatible_provider_info PASSED [ 57%]
tests/llm_translation/test_v0.py::test_v0_in_provider_lists PASSED       [ 71%]
tests/llm_translation/test_v0.py::test_v0_models_configuration PASSED    [ 85%]
tests/llm_translation/test_v0.py::test_v0_supported_params PASSED        [100%]

======================== 7 passed, 2 warnings in 1.82s =========================
```

### Notes
- Default API base URL: `https://api.v0.dev/v1`
- All models support vision/multimodal inputs
- Pricing configured based on v0 documentation
- The v0 provider follows the same pattern as other OpenAI-compatible providers in LiteLLM

colesmcintosh committed:
feat: add v0 provider support to LiteLLM

- Add v0 as a new OpenAI-compatible provider
- Support all three v0 models: v0-1.0-md, v0-1.5-md, v0-1.5-lg
- Configure correct token limits and pricing for each model
- Enable vision support for all v0 models (multimodal)
- Add provider detection for v0/ prefix and api.v0.dev endpoint
- Include comprehensive unit tests for the provider

The v0 provider uses the standard OpenAI-compatible implementation
and supports all standard features including streaming, function
calling, and system messages.

colesmcintosh committed:
Merge branch 'BerriAI:main' into feature/v0-models

colesmcintosh committed:
fix: add v0 provider to ProviderConfigManager

Add V0ChatConfig to the get_provider_chat_config method to fix
test_supports_tool_choice test failure. The v0 provider needs to
be included in the provider config manager to return the correct
configuration for tool choice support detection.

ishaan-jaff reviewed (changes requested):
please add a doc, otherwise LGTM

ishaan-jaff reviewed (changes requested):
please fix list of params

colesmcintosh committed:
docs: add documentation for v0 provider

- Add comprehensive v0 provider documentation
- Cover all supported models and their capabilities
- Include examples for SDK usage, proxy configuration, and all features
- Document supported OpenAI parameters based on v0 API docs
- Add v0 to the providers sidebar navigation

colesmcintosh committed:
fix: correct v0 supported OpenAI parameters

Based on review feedback and v0 API documentation:
- v0 only supports: messages, model, stream, tools, tool_choice
- Remove unsupported parameters like temperature, max_tokens, etc.
- Update tests to verify correct parameter set
- Update documentation to reflect actual API capabilities
- Remove JSON mode example as response_format is not supported

Reference: https://v0.dev/docs/v0-model-api#request-body

colesmcintosh committed:
fix: remove supports_response_schema from v0 models

Remove the supports_response_schema property from all v0 models in the model configuration files as v0 does not support this feature.

Models updated:
- v0/v0-1.0-md
- v0/v0-1.5-md
- v0/v0-1.5-lg

ishaan-jaff reviewed (approved):
LGTM
